### PR TITLE
add auth fallback to docs

### PR DESCRIPTION
**Pull request by alexromano**

### CODE REVIEW

File: docs/Development-Guides/Auth-Fallback.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Auth Fallback
+
+Auth Fallback employers to authorize through Finch Connect even when Finch is experiencing intermittent issues with provider integrations. If an employer attempts to authorize through Finch Connect’s automated flow and receives an error, after three attempts*, Finch Connect will display instructions for the employer to connect Finch manually. This entails the employer adding Finch as a third-party user to the employer’s system. Once Finch has accepted the invite and set up the account, the data for this connection will flow through the Finch API just as it does for every other automated connection. The only key difference is that until the connection is set up on Finch’s side, **your application will receive a 202 response when making requests with that token.**
```
Comment (seii) [inline docs/Development-Guides/Auth-Fallback.md:3]: Suggested: "Auth Fallback allows employers..."

File: docs/Development-Guides/Auth-Fallback.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+For ADP Workforce Now specifically, Finch offers this auth method exclusively. These connections function exactly as above, but rather than being a fallback, there is no fully automated auth method for ADP Workforce Now.
+
+*the number of errors before presenting the fallback screen is configurable. Currently it is set to three for every provider, but we may adjust this value over time.
+
+**Can I opt out?**
+
+Yes, this is an opt-in feature. The tradeoff is that when these intermittent issues do occur, your customers will have no way of connecting their accounts until the issues are resolved.
```
Comment (seii) [inline docs/Development-Guides/Auth-Fallback.md:11]: Suggested: "The tradeoff to keeping the current, non-Fallback, behavior is that..."

File: docs/Development-Guides/Auth-Fallback.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+**Note that if you choose to opt out, Finch’s ADP Workforce now integration not be available to your customers through Finch Connect.**
```
Comment (seii) [inline docs/Development-Guides/Auth-Fallback.md:13]: Suggested: "...if you choose not to opt in..."

(Technically nobody chooses to opt out but can only choose to opt in, based on the wording already present above.)

File: docs/Development-Guides/Auth-Fallback.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+**Why do auth issues occur?**
+
+Auth issues occur for a variety of reasons. The provider backends that Finch is integrating with are often legacy systems, meaning they can often be unreliable. Finch attempts to smooth over these gaps in provider systems with our data sync functionality for serving API requests. This fallback is intended to serve the same purpose for employer authorization.
```
Comment (seii) [inline docs/Development-Guides/Auth-Fallback.md:17]: Suggested: Hyperlink to docs for Data Sync page for the "data sync" phrase.

**Review by seii — CHANGES_REQUESTED** [review]
Mostly good, suggested a few edits for clarity

**Review by seii — APPROVED** [review]

**Comment by alexromano** [comment]
@seii added a new line to call out that this only applies to credential auth

**Review by seii — APPROVED** [review]

**Review by jerzzhang — CHANGES_REQUESTED** [review]
I put everything through ChatGPT for grammar and clarity. See below if you want to replace parts.

Configuring Authentication Methods
Finch offers multiple authentication methods to accommodate various integration preferences of end employers. These methods include credentials-based authentication, OAuth, and direct API token input. Each method provides a distinct balance between reliability and data accessibility.

As a developer, you have the flexibility to decide which authentication methods to make available for your end customers and their order of priority. You can set primary and secondary authentication methods for each provider, and even hide specific methods if desired. For example, if you want to present only the OAuth method to BambooHR customers, Finch can conceal the credentials-based authentication method in Finch Connect. To request a custom authentication configuration, please contact your Finch representative, who will be happy to assist you.

Authentication Fallback Feature
The Authentication Fallback feature is an opt-in option for Scale tier customers. It enables employers to authorize through Finch Connect even during intermittent issues with provider integrations. This feature is applicable only to credentials-based authentication methods and does not extend to direct API token entry or OAuth.

When an employer encounters an error while attempting to authenticate via Finch Connect's automated flow, Finch Connect will display manual connection instructions after three failed attempts*. This process involves the employer adding Finch as a third-party user in their system. Once Finch accepts the invitation and sets up the account, the data for the connection will flow through the Finch API, just like any other automated connection. The primary distinction is that your application will receive a 202 response when making requests with that token until the connection is established on Finch's end. Ensure that your application is configured to handle 202 response codes from Finch. To enable this feature, please reach out to your Finch account manager.

*The number of errors before displaying the fallback screen is adjustable. While the default value is set to three for every provider, this may change over time.

**Pull request merged**